user-management/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+const hashKey = process.env.TOKEN_SECRET ?? '';
+if (!hashKey) throw new Error('TOKEN_SECRET not found in env');
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void {
+  const authHeader = req.headers.authorization;
+  const authToken = authHeader?.split('Bearer ')[1];
+  if (!authHeader || !authToken) {
```

Comment:
This works great, but I thought I'd point out that if `authHeader` is undefined, then `authToken` also will be, so you could just check `authToken`.